tools/template_based_query_generation/src/main/java/parser/Utils.java

```diff
@@ -214,7 +159,7 @@ private static String getOutputDirectory(String directoryName) {
 
     ImmutableMap.Builder<String, ImmutableList<Mapping>> builder = ImmutableMap.builder();
 
-    for (Feature feature : features.getFeatures()) {
+    for (Feature feature : ImmutableList.copyOf(features.getFeatures())) {
```

Comment:
You probably won't need to make a immutable copy of the list here if you are only using it here and you are not modifying the list.

Reply:
Changed, and will look for similar occurrences throughout.